### ISSUE TITLE
Implement the Postgres timezone function

### ISSUE THREAD

**Issue by hawkfish** [duckdb/duckdb#5314]
The [timezone function](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-ZONECONVERT) is what the syntactic sugar of `AT TIME ZONE` translates to but we have not implemented it:

```sql
select now() at time zone 'europe/london' as start_time;
Error: Binder Error: No function matches the given name and argument types 'timezone(VARCHAR, TIMESTAMP)'. You might need to add explicit type casts.
    Candidate functions:
    timezone(TIMESTAMP WITH TIME ZONE) -> BIGINT
    timezone(DATE) -> BIGINT
    timezone(TIMESTAMP) -> BIGINT
    timezone(INTERVAL) -> BIGINT
```

What it seems to do is either a) reinterpret a `TIMESTAMP`'s (UTC) bin values as values in the given zone and use them to produce a `TIMESTAMPTZ` or b) produce a TIMESTAMP whose (UTC) bin values are the same as the bin values of the `TIMESTAMPTZ`'s in the given zone.

Basically, it is a mechanism for encoding local time into instants. Which is sketchy but not hard to implement in ICU.

**Comment by hawkfish** [comment]
I just noticed that we have this function, but it takes one argument and extracts the time zone value (like `year`). the PG API is a two argument function, starting with a `VARCHAR`.
